src/zenml/model/model.py

```diff
@@ -819,3 +830,81 @@ def __hash__(self) -> int:
                 )
             )
         )
+
+    def _prepare_model_version_before_step_launch(
+        self,
+        pipeline_run: "PipelineRunResponse",
+        step_run: Optional["StepRunResponse"],
+        return_logs: bool,
+    ) -> str:
+        """Prepares model version inside pipeline run.
+
+        Args:
+            pipeline_run: pipeline run
+            step_run: step run (passed only if model version is defined in a step explicitly)
+            return_logs: whether to return logs or not
+
+        Returns:
+            Logs related to the Dashboard URL to show later.
+        """
+        from zenml.client import Client
+        from zenml.models import PipelineRunUpdate, StepRunUpdate
+
+        logs = ""
+
+        # copy Model instance to prevent corrupting configs of the
+        # subsequent runs, if they share the same config object
+        self_copy = self.model_copy()
+
+        # in case request is within the step and no self-configuration is provided
+        # try reuse what's in the pipeline run first
+        if step_run is None and pipeline_run.model_version is not None:
```

Comment:
I'm assuming we never get here with a `step_run` that already has a `model_version`, correct?

Reply:
Yep, we hit this code once at the step launch.